### PR TITLE
Migrate all tests to testthat 3e

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR focuses on moving all tests over to the 3rd Edition of **testthat**.

### CODE REVIEW

File: tests/testthat/test-cols_align.R [hunk 1]
```diff
@@ -1,13 +1,12 @@
-context("Ensuring that the `cols_align*()` functions work as expected")
```
Comment (cscheid) [inline tests/testthat/test-cols_align.R:1]: Just flagging that I see a number of these `context()` calls being removed and not added back anywhere.

File: tests/testthat/test-tab_style.R [hunk 2]
```diff
@@ -429,10 +427,7 @@ test_that("a gt table can store the correct style statements", {
 
   # Check that `tab_row_group(others_label = ...)` still works but
   # issues a warning
-  expect_equivalent(
-    expect_warning(data %>% tab_row_group(others_label = "Others1")),
-    data %>% tab_options(row_group.default_label = "Others1")
-  )
+  expect_warning(data %>% tab_row_group(others_label = "Others1"))
```
Comment (cscheid) [inline tests/testthat/test-tab_style.R:430]: The pattern doesn't match the other tests here - I could 100% be wrong but are we missing `expect_equal` or something like that?

File: tests/testthat/test-cols_align.R (hunk 1 above)
Comment (rich-iannone) [reply tests/testthat/test-cols_align.R:1]: The `context()` function is deprecated in `testthat` 3rd edition. For that reason, all are being removed in this migration.

File: tests/testthat/test-tab_style.R (hunk 2 above)
Comment (rich-iannone) [reply tests/testthat/test-tab_style.R:430]: For this case, we really just want to know that there is a warning from a soft deprecation (on our part) and we don't really care about the result.

**Review by cscheid — APPROVED** [review]
lgtm

**Pull request merged**